core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```
@@ -65,219 +74,475 @@
+        EndPoint endpoint = new EndPoint("localhost", 9092,
+                ListenerName.forSecurityProtocol(securityProtocol), securityProtocol);
+        BrokerInfo brokerInfo = BrokerInfo.apply(Broker.apply(id, endpoint,
+                scala.None$.empty()), MetadataVersion.latestTesting(), 9192);
         zkClient.registerBroker(brokerInfo);
     }
 
-    @SafeVarargs
-    static <T> Seq<T> seq(T...seq) {
-        return seq(Arrays.asList(seq));
+    private List<String> generateDefaultAlterOpts(String bootstrapServers) {
+        return asList("--bootstrap-server", bootstrapServers,
+                "--entity-type", "brokers",
+                "--entity-name", "0", "--alter");
+    }
+
+    private void alterAndVerifyConfig(Admin client, Optional<String> brokerId, Map<String, String> config) {
+        alterConfigWithKraft(client, brokerId, config);
+        verifyConfig(client, brokerId, config);
+    }
+
+    private void alterConfigWithKraft(Admin client, Optional<String> brokerId, Map<String, String> config) {
+        String configStr = transferConfigMapToString(config);
+        ConfigCommand.ConfigCommandOptions addOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId), asList("--add-config", configStr)));
+        ConfigCommand.alterConfig(client, addOpts);
     }
 
-    @SuppressWarnings({"deprecation"})
-    static <T> Seq<T> seq(Collection<T> seq) {
-        return JavaConverters.asScalaIteratorConverter(seq.iterator()).asScala().toSeq();
+    private void verifyConfig(Admin client, Optional<String> brokerId, Map<String, String> config) throws RuntimeException {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.BROKER, brokerId.orElse(defaultBrokerId));
+        client.describeConfigs(singletonList(configResource))
+                .all()
+                .whenComplete((c, e) -> {
```

This will swallow the exception since it is running in `whenComplete`